go/types/graph_builder.go

```diff
@@ -112,19 +113,23 @@ type graphOpContainer struct {
 // object.
 func (b *GraphBuilder) Build() Value {
 	var opc opCache
+	var opcStore opCacheStore
 
-	checkFirstCall := func() {
+	defer func() {
+		opcStore.destroy()
+	}()
+
+	// Use function here to take advantage fo the deferred call to mutex.Unlock()
+	func() {
 		b.mutex.Lock()
```

Comment:
Yeah, I guess it is a little surprising but most of this just ends up making local changes to the data and then passing it down to ldb.Put() which is thread safe. I believe this is correct. It gets more complicated when we're pulling stuff out of ldb but that's all protected by this single check.